evm_loader/cli/src/main.rs

```diff
@@ -519,11 +519,43 @@ fn get_ethereum_caller_credentials(
     let caller_ether: H160 = keccak256_h256(&caller_public.serialize()[1..]).into();
     let (caller_sol, caller_nonce) = Pubkey::find_program_address(&[&caller_ether.to_fixed_bytes()], &config.evm_loader);
     let caller_token = spl_associated_token_account::get_associated_token_address(&caller_sol, &evm_loader::token::token_mint::id());
+    let caller_holder = create_block_token_account(config, &caller_ether, &caller_sol).unwrap();
     debug!("caller_sol = {}", caller_sol);
     debug!("caller_ether = {}", caller_ether);
     debug!("caller_token = {}", caller_token);
 
-    (caller_private, caller_ether, caller_sol, caller_nonce, caller_token)
+    (caller_private, caller_ether, caller_sol, caller_nonce, caller_token, caller_holder)
+}
+
+fn create_block_token_account(
+    config: &Config,
+    caller_ether: &H160,
+    caller_sol: &Pubkey,
+) -> Result<Pubkey, Error> {
+    use solana_sdk::program_pack::Pack;
+    let creator = &config.signer;
+    let minimum_balance_for_account = config.rpc_client.get_minimum_balance_for_rent_exemption(spl_token::state::Account::LEN)?;
+    let holder_seed = bs58::encode(&caller_ether.to_fixed_bytes()).into_string() + "hold";
+    let caller_holder = Pubkey::create_with_seed(caller_sol, &holder_seed, &spl_token::id())?;
+    if config.rpc_client.get_account_with_commitment(&caller_holder, CommitmentConfig::confirmed())?.value.is_none() {
+        let instruction = Instruction::new_with_bincode(
+            config.evm_loader,
+            &(4_u32, caller_sol, holder_seed, minimum_balance_for_account, spl_token::state::Account::LEN, spl_token::id(), caller_holder),
```

Comment:
What you trying to achieve?
Anyone can create a token account for anyone else. You don't need to do this inside the evm loader.

Reply:
I didn't manage to create any token apart from associated for caller outside evm_loader.